addon/index.js

```diff
@@ -0,0 +1,62 @@
+import { isEmpty } from '@ember/utils';
+import { get } from '@ember/object';
+import googletag from 'googletag';
+
+
+export function doTargetingForPath(location=window.location) {
+  doTargeting({
+    'host': location.host,
+    'url': location.pathname,
+    'urlSegments': location.pathname && location.pathname.split('/').filter(segment => segment.length > 0)
+  });
+}
+
+export function clearTargetingForPath() {
+  clearTargeting('host');
+  clearTargeting('url');
+  clearTargeting('urlSegments');
+}
+
+export function doTargetingForModels(...models) {
+  models.forEach(model => {
+    let targets = {};
+    let adTargeting = get(model, 'adTargeting');
+    if(adTargeting) {
+      Object.keys(adTargeting).forEach(key => {
+        targets[key] = get(model, adTargeting[key]);
```

Comment:
I head to read this a couple times to understand what's going on, but I know I see it. I can understand why you'd want the key for DFP to not necessarily be the same name as the key on the model (thus coupling the data structures, making an array of simple strings not a great option for `adTargeting`).

But what's the most common case here? I can imagine 80% of the time the model key and DFP key will be the same name. So at the expense of making this logic more complicated, can you refactor so that `adTargeting` can be friendlier for devs? What if `adTargeting` was an array of either strings or objects? Something like this:
```js
['simple-key', {dfpKey: 'modelKey'}, 'otherKey']
```
So for the most common case, if the value in `adTargeting` is a string, use it in both places, but if the value is an object (with a single key), use the key of that object for the DFP key, and use the value as a keyname for lookup on the model.

Reply:
I like this idea. It kind of has echoes of the ES6 shorthand for object properties.

Reply:
Hmm. After thinking more about it I'm not so sure.  Since ads have a flat keyspace, things like `{'story': 'slug'}` and `{'show': 'slug'} like a pretty common use case, and even something like tags might become `{'show-tags': 'tags'}` or `{'story-tags': 'tags'}` depending on how we define the ad spec.

But what really has me leaning the other way is trying to describe it in the docs has gone from explaining a simple key-value map to trying to describe an "array of sometimes a string, sometimes a single-key object". Especially, the single key constrained objects feel weird and bring some new weird edge cases like how to handle an object with multiple keys.

What do you think  about something like how `classNameBindings` worked? Maybe even call it `adBindings`.
```
['modelkey:dfp-key', 'modelkey2', 'modelkey3']
```

Reply:
this approach + adBindings wfm

Reply:
same here. good way to keep it simple